src/IB/IBFEMethod.cpp

```diff
@@ -774,6 +776,15 @@ IBFEMethod::interpolateVelocity(const int u_data_idx,
                                                             data_time,
                                                             /*close_F*/ false,
                                                             /*close_X*/ false);
```

Comment:
This logic is really complicated since `interpWeighted` no longer does weighted interpolation in the nodal quadrature case. Could we change `interpWeighted` to always create a weighted RHS and then just solve for things normally later on so we avoid this kind of problem? I would rather be consistent than maximally efficient.

Reply:
Sure — I just thought of a way to do this without calling close().

However, because nodal interpolation uses insert instead of add, I think we cannot avoid either calling assemble or using INSERT_VALUES at some point, which is different from the other code path when using ghosted assembly.

Reply:
I need to reread our code for setting up a lumped mass matrix but I believe we still need to call 'add' to get the element contributions that are not on the current processor.

Reply:
OK I have reverted this.